test/gtest/cartesianprod.cpp

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "cartesianprod.hpp"
+using float16 = half_float::half;
+
+// FORWARD TEST
+using GPU_CartesianProd_fwd_FP32  = CartesianProdTestFwd<float>;
+using GPU_CartesianProd_fwd_FP16  = CartesianProdTestFwd<float16>;
+using GPU_CartesianProd_fwd_BFP16 = CartesianProdTestFwd<bfloat16>;
+
+TEST_P(GPU_CartesianProd_fwd_FP32, CartesianProdTestFwd)
+{
+    RunTest();
+    Verify();
+};
+
+TEST_P(GPU_CartesianProd_fwd_FP16, CartesianProdTestFwd)
+{
+    RunTest();
+    Verify();
+};
+
+INSTANTIATE_TEST_SUITE_P(Smoke,
+                         GPU_CartesianProd_fwd_FP32,
+                         testing::ValuesIn(CartesianProdTestConfigs()));
+INSTANTIATE_TEST_SUITE_P(Smoke,
+                         GPU_CartesianProd_fwd_FP16,
+                         testing::ValuesIn(CartesianProdTestConfigs()));
```

Comment:
are you missing `INSTANTIATE_TEST_SUITE_P()` for type `bfloat16`? And btw why not use test type of "Full" instead of "Smoke"?